diff --git a/main.m b/main.m
index 51deeb5..13f6687 100644
--- a/main.m
+++ b/main.m
@@ -58,11 +58,9 @@
 #define LIBJLI_DY_LIB "lib/jli/libjli.dylib"
 #define DEPLOY_LIB    "lib/deploy.jar"
 
-//*
-    #define DLog(...) NSLog(@"%s %@", __PRETTY_FUNCTION__, [NSString stringWithFormat:__VA_ARGS__])
-/*/
-    #define DLog(...) do { } while (0)
-//*/
+
+#define DLog(...) NSLog(@"%s %@", __PRETTY_FUNCTION__, [NSString stringWithFormat:__VA_ARGS__])
+
 
 typedef int (JNICALL *JLI_Launch_t)(int argc, char ** argv,
                                     int jargc, const char** jargv,
